File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.java
class Solution {
    public int solution(int a, int b, int c) {
        int ret = 0;
        if(a != b && b != c && c != a){ //3개 서로달라
            ret = a + b + c;
        }else if(a == b && b == c && c == a){ // 3개 서로같아
            ret = (a + b + c)*(a*a + b*b + c*c)*(a*a*a + b*b*b + c*c*c);
        } else { //2개는 같고 1개는 달라
            ret = (a + b + c) * (a*a + b*b + c*c);
        }
        return ret;
    }
}
//??

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.py
from collections import Counter
def solution(a, b, c):
    cnt = Counter([a, b, c])
    if len(cnt) == 3:
        return a + b + c
    elif len(cnt) == 1:
        return (3*a) * (3*(a**2)) * 3*(a**3)
    else:
        return (a+b+c) *(a**2 + b**2 + c**2)